src/coreComponents/physicsSolvers/multiphysics/SinglePhaseReservoirAndWells.hpp

```diff
@@ -13,52 +13,50 @@
  */
 
 /**
- * @file SinglePhaseReservoir.hpp
+ * @file SinglePhaseReservoirAndWells.hpp
  *
  */
 
-#ifndef GEOSX_PHYSICSSOLVERS_MULTIPHYSICS_SINGLEPHASERESERVOIR_HPP_
-#define GEOSX_PHYSICSSOLVERS_MULTIPHYSICS_SINGLEPHASERESERVOIR_HPP_
+#ifndef GEOSX_PHYSICSSOLVERS_MULTIPHYSICS_SINGLEPHASERESERVOIRANDWELLS_HPP_
+#define GEOSX_PHYSICSSOLVERS_MULTIPHYSICS_SINGLEPHASERESERVOIRANDWELLS_HPP_
 
-#include "physicsSolvers/multiphysics/ReservoirSolverBase.hpp"
+#include "physicsSolvers/multiphysics/CoupledReservoirAndWellsBase.hpp"
+#include "physicsSolvers/fluidFlow/wells/SinglePhaseWell.hpp"
 
 namespace geosx
 {
 
-class SinglePhaseReservoir : public ReservoirSolverBase
+template< typename SINGLEPHASE_RESERVOIR_SOLVER >
```

Comment:
Maybe (or maybe not) add a trait to validate that `SINGLEPHASE_RESERVOIR_SOLVER` really is a single phase.

Reply:
I have to find a good way to do that. Would it be something like adding:
```
static bool isSinglePhase() { return "true"; }
```
in `SinglePhaseBase.hpp` and adding 
```
static bool isSinglePhase() { return "false"; }
```
in `CompositionalMultiphaseBase.hpp` or do you mean something else?